hook_macro/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, AttributeArgs};
+use quote::ToTokens;
+
+
+#[proc_macro_attribute]
+/// This macro is used to insert a hook at the start of a function.
+pub fn cvt_hook_start(attr: TokenStream, input: TokenStream) -> TokenStream {
+    
+    // parse the attribute argument
+    let attr = parse_macro_input!(attr as AttributeArgs);
+    if attr.len() != 1 {
+        panic!("expected 1 argument");
```

Comment:
this does not seem like the best error handling. There should be another but simple way to return compile time error from an attribute

Reply:
what do you mean here?

Reply:
You panic rather than return an appropriate error message. My thinking was that `panic` will just break the compilation and make the compiler exit immediately instead of providing a graceful error message

Reply:
https://docs.rs/syn/latest/syn/struct.Error.html

Also seems to use `let input = parse_macro_input!(input as ItemFn);` syntax to parse an expected function.